src/v/cloud_storage/remote.h

```diff
@@ -21,8 +21,26 @@
 #include <seastar/core/gate.hh>
 #include <seastar/core/loop.hh>
 
+#include <utility>
+
 namespace cloud_storage {
 
+/// \brief Predicate required to continue operation
+///
+/// Describes a predicate to be evaluated before starting an expensive
+/// operation, or to be evaluated when an operation has failed, to find
+/// the reason for failure
+struct should_continue_operation {
```

Comment:
The name of this struct seems a bit too general. For example, it's possible to interpret it as if a worker should call `should_continue_operation` to check if any work left and when the methods returns false the worker just successfully return control flow. While the true meaning of this method to abort the worker, to make it return failure and to provide the context for logging. The former interpretation is about normal workflow while the latter is about exceptional workflow. It's a big difference.

The exceptional interpretation is similar to 

 - CancellationToken in .Net - https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken?view=net-6.0
 - abort_source in http://docs.seastar.io/master/classseastar_1_1abort__source.html

However, it isn't exactly the same: abort_source and CancellationToken are explicitly set to interrupt the operation (eager) while `should_continue_operation` polls this information (lazy). We may highlight this peculiarity in the name. What do you think of renaming `should_continue_operation` to `lazy_abort_source`?

Reply:
I have refactored to `lazy_abort_source`, please review